format/spec.md

```diff
@@ -193,10 +193,38 @@ Notes:
+| **`timestamp`**    | **`number`**  | 1646277378000000                       |                                                                                                                                                                                                                                              |
+| **`timestamptz`**  | **`number`**  | 1646277378000000                       |                                                                                                                                                                                                                                              |
+| **`string`**       | **`string`**  | "foo"                                  |                                                                                                                                                                                                                                              |
+| **`uuid`**         | **`string`**  | "eb26bdb1-a1d8-4aa6-990e-da940875492c" |                                                                                                                                                                                                                                              |
+| **`fixed(L)`**     | **`string`**  | "0x3162"                               | we use hexadecimal byte literals to encode bytes, with prefix `0x`                                                                                                                                                                           |
+| **`binary`**       | **`string`**  | "0x3162"                               | we use hexadecimal byte literals to encode bytes, with prefix `0x`                                                                                                                                                                           |
+| **`struct`**       | **`object`**  | {"a": 1, "foo": "bar"}                 | the keys are the nested fields' names in the struct schema, and the values should be corresponding value literals of the field's type                                                                                                        |
+| **`list`**         | **`array`**   | \[1, 2, 3\]                            | each value should be value literals of the corresponding element type of the list                                                                                                                                                            |
+| **`map`**          | **`object`**  | {"a": 1, "b": 2}                       | the key and value type correspond with the schema of the map, the key (json string) are parsed into the resulting key type of the map schema, an exception will be thrown if parsing fails.                                                  |
```

Comment:
This encoding doesn't work for non-string keys. I recommend using an array of arrays that is interpreted as a list of key/value pairs.

Reply:
Yes, a json object's key can only be string type, that's why I mentioned the string key will be parsed into corresponding type. But it seems using array of arrays can preserve/enforce the key type information better.

Reply:
Yeah, JSON within JSON within JSON (etc.) is going to be painful. Let's use a different format.

Reply:
I agree this would make a very difficult situation, especially for a default struct with a map inside of it. I think we should keep everything as default parsable json as possible.